morphia/src/test/java/org/mongodb/morphia/TestDocumentValidation.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2016 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mongodb.morphia;
+
+import com.mongodb.WriteConcernException;
+import org.junit.Test;
+import org.mongodb.morphia.entities.DocumentValidation;
+
+import java.util.Date;
+
+import static org.junit.Assert.assertTrue;
+
+public class TestDocumentValidation extends TestBase {
```

Comment:
You could test that the validation action is applied properly by creating an annotation where the action is WARN, and then assert that an invalid document is still inserted.

You could test validation level is applied properly by creating an annotation where the level is MODERATE and then assert that an already invalid document is not validated on update.

But it's sufficient to assert that the values for validationLevel and validationAction  in the options field from collection info matches the ones (for both the create collection and the collMod cases)